analyzers/src/SonarAnalyzer.CSharp/Rules/InheritedCollidingInterfaceMembers.cs

```diff
@@ -168,8 +168,8 @@ private static bool AreCollidingMethods(IMethodSymbol methodSymbol1, IMethodSymb
                 var param1 = methodSymbol1.Parameters[i];
                 var param2 = methodSymbol2.Parameters[i];
 
-                if (param1.RefKind != param2.RefKind ||
-                    !object.Equals(param1.Type, param2.Type))
+                if (param1.RefKind != param2.RefKind
+                    || !Equals(param1.Type, param2.Type))
```

Comment:
Educational: Is there a reason why  it is not `!param1.Type.Equals(param2.Type)` ?

Reply:
I think `==`, `Equals(x1,x2)` and `x1.Equals(x2)` get resolved differently in general, meaning:

- operator `==`: if implemented, calls the override `==` operator of the type. If not, calls `Equals/ReferenceEquals` on value/reference types respectively.
- `Equals(x1,x2)` calls the default object comparison, which is again `Equals/ReferenceEquals` on value/reference types respectively.
- `x1.Equals(x2)` calls one of the following, in a first-found manner:
    - An implementation of `IEquatable<T>`'s `Equals(T other)`
    - An implementation of `override bool Equal(object other)`
    - `object.Equals(obj other)` if nothing else is available

Now, on this specific case, i think since an ancestor type implements `IEquatable<ISymbol>`, it would change the dispatch of the method to that one if we called `!param1.Type.Equals(param2.Type)`

Most cases that would be the case, with a slight performance hit, but if Microsoft has implemented ISymbol's equatability method in a different manner, the logic might also change.

How much of this is wrong? :)

Reply:
It is not wrong, however, I think it is calling the IEquatable.Equals the way to go if it is already implemented. WDYT?

Reply:
I will merge the PR as is, please take a look at this @gregory-paidis-sonarsource when you are back from Holiday.